### PR TITLE
Allow disabling text shadow in `EditBox`

### PR DESCRIPTION
**Pull request by IchHabeHunger54**
Adds a `shadow` boolean field to `EditBox`, allowing users to disable the text shadow for textboxes if needed. The value defaults to true, mirroring vanilla behavior.

Previously, my solution for this problem was to [subclass `EditBox` and reimplement rendering myself](https://github.com/MinecraftschurliMods/Bibliocraft-Legacy/blob/1ca02a8db3698629afe5d8170abb7100189361f7/src/main/java/com/github/minecraftschurlimods/bibliocraft/client/screen/ClipboardScreen.java#L96-L143). I would like to avoid having to do that, plus I imagine this change might be useful to others, so here we go.

### CODE REVIEW

**Comment by Shadows-of-Fire** [comment]
Needs to be re-targeted against 1.21 (or a new PR has to be opened with this as a backport)

File: patches/net/minecraft/client/gui/components/EditBox.java.patch [hunk 1]
```diff
@@ -0,0 +1,60 @@
+--- a/net/minecraft/client/gui/components/EditBox.java
++++ b/net/minecraft/client/gui/components/EditBox.java
+@@ -56,6 +_,7 @@
+     @Nullable
+     private Component hint;
+     private long focusedTime = Util.getMillis();
++    private boolean shadow = true;
```
Comment (Technici4n) [inline patches/net/minecraft/client/gui/components/EditBox.java.patch]: Would it be a good idea to add a comment explaining why we have this patch? Something along the lines of `// Neo: Allow modders to disable the default shadow.` 🤔

File: patches/net/minecraft/client/gui/components/EditBox.java.patch [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+     public void setHint(Component p_259584_) {
+         this.hint = p_259584_;
++    }
++
++    public void setShadow(boolean shadow) {
++        this.shadow = shadow;
++    }
++
++    public boolean getShadow() {
++        return this.shadow;
```
Comment (ChampionAsh5357) [inline patches/net/minecraft/client/gui/components/EditBox.java.patch]: I would call this `textShadow` since shadow is not necessarily clear.

**Comment by TelepathicGrunt** [comment]
@IchHabeHunger54 Do you want to update this PR for 1.21? Or is this not needed anymore?

**Comment by IchHabeHunger54** [comment]
Yes I do, however NeoDev isn't too eager to work for me atm. Will check again tomorrow.

**Comment by TelepathicGrunt** [comment]
Updated this PR to 1.21 with the suggestions above applied. (I probably should've rebased instead of merging in 1.21 branch...) Here's a test example of me disabling text shadow for world creation screen name box.
image

**Review by XFactHD — APPROVED** [review]

**Review by Technici4n — APPROVED** [review]

**Comment by Technici4n** [comment]
@ChampionAsh5357 do you want to re-review or can we merge?

File: patches/net/minecraft/client/gui/components/EditBox.java.patch [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+ 
+     public void setHint(Component p_259584_) {
+         this.hint = p_259584_;
++    }
++
++    public void setTextShadow(boolean textShadow) {
++        this.textShadow = textShadow;
++    }
++
++    public boolean getTextShadow() {
```
Comment (shartte) [inline patches/net/minecraft/client/gui/components/EditBox.java.patch:57]: Shouldn't this be `isTextShadow` by JavaBean conventions?
p.s.: Javadoc

File: patches/net/minecraft/client/gui/components/EditBox.java.patch [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                 }
+             }
+ 
+@@ -552,5 +_,13 @@
+ 
+     public void setHint(Component p_259584_) {
+         this.hint = p_259584_;
++    }
++
++    public void setTextShadow(boolean textShadow) {
```
Comment (shartte) [inline patches/net/minecraft/client/gui/components/EditBox.java.patch:53]: Needs Javadoc

**Review by ChampionAsh5357 — APPROVED** [review]

**Review by TelepathicGrunt — APPROVED** [review]

**Pull request merged**